flask/src/main.py

```diff
@@ -313,102 +426,211 @@ def get_api_request(key, delay):
 def api():
+    logger.info('Received /api endpoint request', extra={
+        'endpoint': '/api',
+        'method': 'GET'
+    })
     return "flask /api"
 
 
 @app.route('/organization', methods=['GET'])
 @cache.cached(timeout=1000, key_prefix="flask.cache.organization")
 def organization():
+    logger.info('Received /organization endpoint request', extra={
+        'endpoint': '/organization',
+        'method': 'GET'
+    })
+
     # perform get_products db query 1% of time in order
     #   to populate "Found In" endpoints in Queries
     if random.random() < 0.01:
+        logger.info('Processing /organization - executing random products query', extra={
+            'probability_threshold': 0.01
+        })
         rows = get_products()
     return "flask /organization"
 
 
 @app.route('/connect', methods=['GET'])
 def connect():
+    logger.info('Received /connect endpoint request', extra={
+        'endpoint': '/connect',
+        'method': 'GET'
+    })
     return "flask /connect"
 
 
 @app.route('/showSuggestion', methods=['GET'])
 def showSuggestion():
-  return jsonify({"response":os.getenv("OPENAI_API_KEY") is not None}), 200
+    logger.info('Received /showSuggestion endpoint request', extra={
+        'endpoint': '/showSuggestion',
+        'method': 'GET'
+    })
+
+    has_openai_key = os.getenv("OPENAI_API_KEY") is not None
+    logger.info('Processing /showSuggestion - OpenAI key availability checked', extra={
+        'has_openai_key': has_openai_key
+    })
+
+    return jsonify({"response": has_openai_key}), 200
 
 @app.route('/product/0/info', methods=['GET'])
 def product_info():
+    logger.info('Received /product/0/info endpoint request', extra={
+        'endpoint': '/product/0/info',
+        'method': 'GET',
+        'sleep_duration': 0.55
+    })
+
     time.sleep(.55)
+    logger.info('Completed /product/0/info request')
     return "flask /product/0/info"
 
 
 # uncompressed assets
 @app.route('/uncompressed_assets/<path:path>')
 def send_report(path):
+    logger.info('Received /uncompressed_assets request', extra={
+        'endpoint': '/uncompressed_assets',
+        'file_path': path,
+        'sleep_duration': 0.55
+    })
+
     time.sleep(.55)
     response = send_from_directory('../uncompressed_assets', path)
     # `Timing-Allow-Origin: *` allows timing/sizes to visbile in span
     response.headers['Timing-Allow-Origin'] = '*'
     # Overwriting `Content-Type` header to disable compression
     response.headers['Content-Type'] = 'application/octet-stream'
+
+    logger.info('Completed /uncompressed_assets request', extra={
+        'file_path': path,
+        'content_type': 'application/octet-stream'
+    })
+
     return response
 
 
 # compressed assets
 @app.route('/compressed_assets/<path:path>')
 def send_report_configured_properly(path):
+    logger.info('Received /compressed_assets request', extra={
+        'endpoint': '/compressed_assets',
+        'file_path': path
+    })
```

Comment:
This adds 45 multi-line logging statements. I personally don't love having to wade through all this while maintaining the app.

Q1: do we really need all of them? What's the demo flow we are catering for here? I think we may want some log line in most spans in `/products` in case somebody goes around randomly clicking in the waterfall. Then maybe another 3-5 in `/checkout`. Plus whatever you find useful to actually debug the application (e.g. caches), or anything that's "close" to some demo path, e.g. N+1 query.

Q2: can we make them less verbose? One way is [something like this](https://github.com/getsentry/sentry-helper-slack-bot/blob/547f8629173c4125ade262261adc364d3bf20039/app.js#L297-L299). In a real application this is a must IMO unless you want your code to be 50% instrumentation.